### ISSUE TITLE
Reproduce BERT and TUPE

### ISSUE THREAD

**Issue by wyu97** [guolinke/TUPE#11]
Hi authors,

If I also want to reproduce BERT by using your code. Do I only have to change [`fairseq/modules/transformer_sentence_encoder.py`](fairseq/modules/transformer_sentence_encoder.py) and [`fairseq/modules/multihead_attention.py`](fairseq/modules/multihead_attention.py) back to the original code in fairseq? Do you have other modifications on other files?

Thank you!

**Comment by guolinke** [comment]
yeah, you basically only need to add a bias to the attention correction matrix.

**Comment by wyu97** [comment]
Thanks for your response! 

I am also wondering how many percentage of data in bookcorpus and wikipedia do you calculate the validation loss (Figure 5(a) in the paper) when pre-training? For example, fist 95% bookcorpus for training, last 5% for validation?

**Comment by guolinke** [comment]
you can refer to this code: https://github.com/guolinke/TUPE/blob/master/preprocess/pretrain/split.py#L14
The train: valid is about 199: 1.
